x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/sort/IntBucketedSort.java

```diff
@@ -257,19 +258,23 @@ private void swap(long lhs, long rhs) {
 
     /**
      * Allocate storage for more buckets and store the "next gather offset"
-     * for those new buckets.
+     * for those new buckets. We always grow the storage by whole bucket's
+     * worth of slots at a time. We never allocate space for partial buckets.
      */
-    private void grow(long minSize) {
+    private void grow(int bucket) {
         long oldMax = values.size();
-        values = bigArrays.grow(values, minSize);
+        assert oldMax % bucketSize == 0;
+
+        long newSizeInBuckets = BigArrays.overSize(bucket + 1, PageCacheRecycler.INT_PAGE_SIZE, Integer.BYTES * bucketSize);
+        values = bigArrays.resize(values, newSizeInBuckets * bucketSize);
```

Comment:
I may not be familiar enough with these methods, but the call to `BigArrays.oversize` looks weird.

Inside of `oversize`, I found checks like
```
        if (minTargetSize < pageSize) {
```
which seems to indicate that this method expects the `minTargetSize` to be in the same unit as the page size. Since the given pageSize is `PageCacheRecycler.INT_PAGE_SIZE`, this indicates that the `minTargetSize` should be in multiples of ints that we want to store. That'd be `(bucket + 1)*bucketSize`, not `bucket +1`, no? And then we'd also have to adjust the `bytesPerElement` to be just `Integer.BYTES`.

Also, in the code for `oversize`, it seems like the `bytesPerElement` can only by 1-8. Which would make our `Integer.BYTES * bucketSize` completely off the charts.

Reply:
fun. I looked at it and it seemed to be working. but I do think it'd be clearer if it just did it in bytes always. I'll switch it.

Reply:
I'm going to do what Alex asked for and he's asked me to merge once CI is happy about it - I'm using primitive sizes and rounding up.